src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1523,19 +1537,18 @@ uint_t MPS::apply_measure_internal_single_qubit(uint_t qubit, RngEngine &rng,
   }
   apply_matrix_internal(qubits_to_update, measurement_matrix);
   if (num_qubits_ > 1)
-    propagate_to_neighbors_internal(qubit, qubit, measure_all);
+    propagate_to_neighbors_internal(qubit, qubit, measure_right_neighbor);
 
   return measurement;
 }
 
-void MPS::propagate_to_neighbors_internal(uint_t min_qubit, uint_t max_qubit, 
-					  bool measure_all) {
+void MPS::propagate_to_neighbors_internal(uint_t min_qubit, uint_t max_qubit,
+					  bool measure_right_neighbor) {
   uint_t right_qubit=num_qubits_-1;
   int_t left_qubit=0;
 
-  if (measure_all) {
-    right_qubit = max_qubit >= num_qubits_-2 ? num_qubits_-1 : max_qubit + 1;
-    left_qubit = min_qubit == 0 ? 0 : min_qubit - 1;
+  if (measure_right_neighbor) {
+    right_qubit = (max_qubit >= num_qubits_-2) ? num_qubits_-1 : max_qubit + 1;
```

Comment:
According to this: if we measure says qubits 1, 3, 5 in a circuit of 10 qubits, then after measuring qubit 1 the propagation will carry on all the way up to the 10th qubit, then again when we measure qubit 3 and qubit 5. I thought we wanted to stop at qubit 3 when we measure qubit 1 and at qubit 5 when we measure qubit 3. So, instead of passing the Boolean `measure_right_qubit`, we have to pass the integer `sorted_qubits[i+1]`.

Reply:
You brought up a good point. What you suggest is better and also cleaner. I made several changes based on this, and discovered a couple of failures on the way, so corrected them as well. I will be happy for another review. There is currently a failure in the tests, but I think it is not related to this PR as I am getting it in `main` as well, so need to investigate it there.